server-setup/tasks-handlers.ts

```diff
@@ -81,13 +101,50 @@ const handlers = {
       exitCode === 0 ? "0" : "8"
     );
   },
+
+  "upload-terminology-plugin": async ({ server, task, plugin }: Handler) => {
+    console.log(
+      `📤 Ingesting code system ${task.id} version ${server.version} from ${task.source}...`
+    );
+
+    const dataSource = task.source;
+    const dataType = task.id;
+    const dataVersion = server.version;
+
+    if (
+      await isTerminologySourceImported(server.url, {
+        source: dataSource,
+        system: dataType,
+        version: dataVersion,
+      })
+    ) {
+      console.log(
+        `ℹ️ Code system ${task.id} version ${server.version} from ${task.source} already ingested`
+      );
+      return;
+    }
+
+    const exitCode = (await plugin?.uploadTerminology(server, task)) ?? "8";
+
+    createApplicationImportAuditEvent(
+      server.url,
+      {
+        source: dataSource,
+        system: dataType,
+        version: dataVersion,
+      },
+      exitCode
+    );
+  },
 };
 
 export async function handleTask(server: ConfigServer, task: ConfigTaskEntry) {
   console.log(`▶️ Handling task of type ${task.type}...`);
   const handler = handlers[task.type];
+  const plugin = plugins.find((p) => p.name === task.plugin);
+
   if (handler) {
-    return handler(server, task);
+    return handler({ server, task, plugin });
```

Comment:
why is it finding plugins for other handlers than `upload-terminology-plugin`?
why would other handler types than `upload-terminology-plugin` need to receive a plugin as input?

Reply:
✅